[WPE] Split WPEWebView into Legacy and Platform implementations

carlosgcampos opened the pull request:
#### 00b3d6a56f5149245b9e1501389ae169846ec240
<pre>
[WPE] Split WPEWebView into Legacy and Platform implementations
<a href="https://bugs.webkit.org/show_bug.cgi?id=275985">https://bugs.webkit.org/show_bug.cgi?id=275985</a>

Reviewed by Nikolas Zimmermann.

It makes clearer what code is expected to be used for new and old APIs.

* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/C/wpe/WKView.cpp:
(WKViewCreate):
(WKViewCreateDeprecated):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewCreatePage):
(webkitWebViewGetRendererBufferFormat):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::doneWithTouchEvent):
(WebKit::PageClientImpl::enterAcceleratedCompositingMode):
(WebKit::PageClientImpl::exitAcceleratedCompositingMode):
(WebKit::PageClientImpl::updateAcceleratedCompositingMode):
(WebKit::PageClientImpl::enterFullScreen):
(WebKit::PageClientImpl::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::View::View):
(WKWPE::View::close):
(WKWPE::View::accessible const):
(): Deleted.
(WKWPE::m_backend): Deleted.
(WKWPE::View::~View): Deleted.
(WKWPE::View::setViewState): Deleted.
(WKWPE::View::handleKeyboardEvent): Deleted.
(WKWPE::View::synthesizeCompositionKeyPress): Deleted.
(WKWPE::viewToplevelIsFullScreen): Deleted.
(WKWPE::View::enterFullScreen): Deleted.
(WKWPE::View::didEnterFullScreen): Deleted.
(WKWPE::View::exitFullScreen): Deleted.
(WKWPE::View::didExitFullScreen): Deleted.
(WKWPE::View::requestExitFullScreen): Deleted.
(WKWPE::View::setFullScreen): Deleted.
(WKWPE::View::platformWebPageProxyForGamepadInput): Deleted.
(WKWPE::View::updateAcceleratedSurface): Deleted.
(WKWPE::View::renderBufferFormat const): Deleted.
(WKWPE::View::updateDisplayID): Deleted.
(WKWPE::View::touchPointsForEvent): Deleted.
(WKWPE::View::setCursor): Deleted.
(WKWPE::View::callAfterNextPresentationUpdate): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
(WKWPE::View::page):
(WKWPE::View::client const):
(WKWPE::View::size const):
(WKWPE::View::backend const):
(WKWPE::View::wpeView const):
(WKWPE::View::setCursor):
(WKWPE::View::create): Deleted.
(WKWPE::View::backend): Deleted.
(WKWPE::View::touchGestureController const): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.cpp: Added.
(WKWPE::ViewLegacy::ViewLegacy):
(WKWPE::ViewLegacy::~ViewLegacy):
(WKWPE::ViewLegacy::setViewState):
(WKWPE::ViewLegacy::handleKeyboardEvent):
(WKWPE::ViewLegacy::synthesizeCompositionKeyPress):
(WKWPE::ViewLegacy::setFullScreen):
(WKWPE::ViewLegacy::platformWebPageProxyForGamepadInput):
(WKWPE::ViewLegacy::callAfterNextPresentationUpdate):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.h: Added.
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.cpp: Added.
(WKWPE::ViewPlatform::ViewPlatform):
(WKWPE::ViewPlatform::~ViewPlatform):
(WKWPE::viewToplevelIsFullScreen):
(WKWPE::ViewPlatform::enterFullScreen):
(WKWPE::ViewPlatform::didEnterFullScreen):
(WKWPE::ViewPlatform::exitFullScreen):
(WKWPE::ViewPlatform::didExitFullScreen):
(WKWPE::ViewPlatform::requestExitFullScreen):
(WKWPE::ViewPlatform::updateAcceleratedSurface):
(WKWPE::ViewPlatform::renderBufferFormat const):
(WKWPE::ViewPlatform::updateDisplayID):
(WKWPE::ViewPlatform::touchPointsForEvent):
(WKWPE::ViewPlatform::synthesizeCompositionKeyPress):
(WKWPE::ViewPlatform::setCursor):
(WKWPE::ViewPlatform::callAfterNextPresentationUpdate):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.h: Added.
* Source/WebKit/UIProcess/Gamepad/libwpe/UIGamepadProviderLibWPE.cpp:
(WebKit::UIGamepadProvider::platformWebPageProxyForGamepadInput):

Canonical link: <a href="https://commits.webkit.org/280522@main">https://commits.webkit.org/280522@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/926d53c72fc872286bcbd01d6488295e08c31290

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56813 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36141 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60429 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7256 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43764 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7446 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46014 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5079 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58843 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33950 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49028 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26871 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30730 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6260 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52665 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6630 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62111 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/726 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6734 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53272 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/729 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49083 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53294 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/612 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8468 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31970 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33055 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34140 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32801 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
Quite an improvment, r=me!

carlosgcampos committed:
[WPE] Split WPEWebView into Legacy and Platform implementations
https://bugs.webkit.org/show_bug.cgi?id=275985

Reviewed by Nikolas Zimmermann.

It makes clearer what code is expected to be used for new and old APIs.

* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/C/wpe/WKView.cpp:
(WKViewCreate):
(WKViewCreateDeprecated):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewCreatePage):
(webkitWebViewGetRendererBufferFormat):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::doneWithTouchEvent):
(WebKit::PageClientImpl::enterAcceleratedCompositingMode):
(WebKit::PageClientImpl::exitAcceleratedCompositingMode):
(WebKit::PageClientImpl::updateAcceleratedCompositingMode):
(WebKit::PageClientImpl::enterFullScreen):
(WebKit::PageClientImpl::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::View::View):
(WKWPE::View::close):
(WKWPE::View::accessible const):
(): Deleted.
(WKWPE::m_backend): Deleted.
(WKWPE::View::~View): Deleted.
(WKWPE::View::setViewState): Deleted.
(WKWPE::View::handleKeyboardEvent): Deleted.
(WKWPE::View::synthesizeCompositionKeyPress): Deleted.
(WKWPE::viewToplevelIsFullScreen): Deleted.
(WKWPE::View::enterFullScreen): Deleted.
(WKWPE::View::didEnterFullScreen): Deleted.
(WKWPE::View::exitFullScreen): Deleted.
(WKWPE::View::didExitFullScreen): Deleted.
(WKWPE::View::requestExitFullScreen): Deleted.
(WKWPE::View::setFullScreen): Deleted.
(WKWPE::View::platformWebPageProxyForGamepadInput): Deleted.
(WKWPE::View::updateAcceleratedSurface): Deleted.
(WKWPE::View::renderBufferFormat const): Deleted.
(WKWPE::View::updateDisplayID): Deleted.
(WKWPE::View::touchPointsForEvent): Deleted.
(WKWPE::View::setCursor): Deleted.
(WKWPE::View::callAfterNextPresentationUpdate): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebView.h:
(WKWPE::View::page):
(WKWPE::View::client const):
(WKWPE::View::size const):
(WKWPE::View::backend const):
(WKWPE::View::wpeView const):
(WKWPE::View::setCursor):
(WKWPE::View::create): Deleted.
(WKWPE::View::backend): Deleted.
(WKWPE::View::touchGestureController const): Deleted.
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.cpp: Added.
(WKWPE::ViewLegacy::ViewLegacy):
(WKWPE::ViewLegacy::~ViewLegacy):
(WKWPE::ViewLegacy::setViewState):
(WKWPE::ViewLegacy::handleKeyboardEvent):
(WKWPE::ViewLegacy::synthesizeCompositionKeyPress):
(WKWPE::ViewLegacy::setFullScreen):
(WKWPE::ViewLegacy::platformWebPageProxyForGamepadInput):
(WKWPE::ViewLegacy::callAfterNextPresentationUpdate):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.h: Added.
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.cpp: Added.
(WKWPE::ViewPlatform::ViewPlatform):
(WKWPE::ViewPlatform::~ViewPlatform):
(WKWPE::viewToplevelIsFullScreen):
(WKWPE::ViewPlatform::enterFullScreen):
(WKWPE::ViewPlatform::didEnterFullScreen):
(WKWPE::ViewPlatform::exitFullScreen):
(WKWPE::ViewPlatform::didExitFullScreen):
(WKWPE::ViewPlatform::requestExitFullScreen):
(WKWPE::ViewPlatform::updateAcceleratedSurface):
(WKWPE::ViewPlatform::renderBufferFormat const):
(WKWPE::ViewPlatform::updateDisplayID):
(WKWPE::ViewPlatform::touchPointsForEvent):
(WKWPE::ViewPlatform::synthesizeCompositionKeyPress):
(WKWPE::ViewPlatform::setCursor):
(WKWPE::ViewPlatform::callAfterNextPresentationUpdate):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.h: Added.
* Source/WebKit/UIProcess/Gamepad/libwpe/UIGamepadProviderLibWPE.cpp:
(WebKit::UIGamepadProvider::platformWebPageProxyForGamepadInput):

Canonical link: https://commits.webkit.org/280522@main
